src/dash-table/components/Filter/Column.tsx

```diff
@@ -40,22 +46,51 @@ export default class ColumnFilter extends PureComponent<IColumnFilterProps, ISta
         const {
             classes,
             columnId,
+            columns,
             isValid,
             style,
-            value
+            value,
+            globalFilterCase,
+            columnFilterCaseSensitive,
+            columnFilterCaseInsensitive,
+            setProps
         } = this.props;
 
+        const filterCaseClass: string =
+            (globalFilterCase !== Case.Insensitive && !columnFilterCaseInsensitive) ?
+                'dash-filter--case--sensitive' : 'dash-filter--case--insensitive';
+
+        function setColumnCase() {
+            const cols: IColumn[] = R.clone(columns);
+            const inx: number = R.findIndex(R.propEq('id', columnId))(cols);
+
+            cols[inx].filter_case_sensitive = !columnFilterCaseSensitive &&
+                globalFilterCase === Case.Insensitive;
+            cols[inx].filter_case_insensitive = !columnFilterCaseInsensitive &&
+                (globalFilterCase === Case.Sensitive || globalFilterCase === Case.Default);
+
+            setProps({ columns: cols });
+        }
+
         return (<th
             className={classes + (isValid ? '' : ' invalid')}
             data-dash-column={columnId}
             style={style}
         >
             <IsolatedInput
                 value={value}
-                placeholder={`filter data...`}
+                placeholder='filter data...'
                 stopPropagation={true}
                 submit={this.submit}
             />
+            <div>
+                <input
+                    type='button'
+                    className={'dash-filter--case ' + filterCaseClass}
+                    onClick={setColumnCase}
+                    value='Aa'
+                />
```

Comment:
Had to make this an `input` type button, because there was no other way to make this thing have `cursor: pointer`, it would still be `text` from `input` below it. Setting `z-index` did nothing.